### PR TITLE
vet_clinic: Update animals table and add queries

### PR DESCRIPTION
**Pull request by yodit93**
### In this project, I

- [ ] added a column species of type string to animals table,

- [ ] inserted new data,

- [ ] Inside a transaction

-     updated the animals table by setting the species column to unspecified,

image

-     updated the animals table by setting the species column to digimon for all animals that have a name ending in mon,
   
image

-     updated the animals table by setting the species column to pokemon for all animals that don't have species already set,

image

-     committed the transaction and verified that the change was made and persists after the commit
  
image

-     deleted all records in the animals table,

    

-     rolled back the transaction and verified all records in the animals table exists.

image

-     deleted all animals born after Jan 1st, 2022,
   
image

-     created a savepoint for the transaction. and updated animals' weights to be their weight multiplied by -1.

image

-     rolled back to the savepoint, updated all animals' weights that are negative to be their weight multiplied by -1 and committed transaction,

  image

- [ ] wrote queries to answer the following questions:
  
   - How many animals are there?

    
count 1

    - How many animals have never tried to escape?
    
count 2

    - What is the average weight of animals?
    
count 3

    - Who escapes the most, neutered or not neutered animals?
    
count 4

    - What is the minimum and maximum weight of each type of animal?
    
count 5

    - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
    
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+VALUES ('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Charmander', '2020-02-08', 0, false, 11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Plantmon', '2021-11-15', 2, true, 5.7);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Squirtle', '1993-04-02', 3, false, 12.13);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Angemon', '2005-06-12', 1, true, 45);
```
Comment (muneebulrehman) [inline data.sql]: The weight of these animals as mentioned is of the negative values. Kindy update your data accordingly. 
image

File: schema.sql [hunk 2]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals (
     date_of_birth date,
     escape_attempts int,
     neutered boolean,
-    weight_kg decimal(5,2)
+    weight_kg decimal(5,2),
+    species VARCHAR(100)
```
Comment (muneebulrehman) [inline schema.sql]: - Optional
Kindy use the `Alter` command to add the species column to the animals table.

File: queries.sql [hunk 3]
```diff
@@ -18,4 +18,49 @@ SELECT * FROM animals WHERE neutered = true;
...
+--How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+--What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+--Who escapes the most, neutered or not neutered animals?
+SELECT neutered, count(*) AS escape_count FROM animals GROUP BY neutered;
+--What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals GROUP BY species;
+--What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(weight_kg) FROM animals WHERE EXTRACT(YEAR FROM date_of_birth) BETWEEN 1990 AND 2000 GROUP BY species;
```
Comment (muneebulrehman) [inline queries.sql]: We need to check for the average escape attempts and not for the average weight as per the requirements. Kindly fix your query accordingly. ⬇️
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @yodit93 , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (yodit93) [reply data.sql]: Ohh....., I didn't consider that. I will update it. Thanks!

File: schema.sql (hunk 2 above)
Comment (yodit93) [reply schema.sql]: Yes, I thought like that before. But the instruction said like this...
"Add a column species of type string to your animal's table. Modify your schema.sql file.".....that is why I just added it on schema.sql manually.
Actually, I used ALTER to insert a new column to the table on pgADMIN. But not on VS code.
I will update it.
Thank you very much!!!

File: queries.sql (hunk 3 above)
Comment (yodit93) [reply queries.sql]: Yes, you are right. I did a mistake. I will correct it.
Thanks.

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @yodit93 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- You successfully implemented the requested changes.✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**